### ISSUE TITLE
Question: Why are the charges not learned using flow-matching?

### ISSUE THREAD

**Issue by mattrmd** [rssrwn/semla-flow#7]
The paper states that atom positions and types and inter-atom bonds are noised, but I see no indication that **charges** are noised (either in the paper or the code):

<img width="775" height="244" alt="Image" src="https://github.com/user-attachments/assets/1b919e8a-630c-40ed-b18f-1b3eef63fb92" />

https://github.com/rssrwn/semla-flow/blob/3f43103d3af138b86dbe9f29fe8085e83f9a6283/semlaflow/data/interpolate.py#L287-L327

Instead, the charges seem to be set to zero by the default behavior of the constructor at every noise level.

https://github.com/rssrwn/semla-flow/blob/3f43103d3af138b86dbe9f29fe8085e83f9a6283/semlaflow/util/molrepr.py#L295

Additionally, the charges seem to be ignored during the flow process:

https://github.com/rssrwn/semla-flow/blob/3f43103d3af138b86dbe9f29fe8085e83f9a6283/semlaflow/models/fm.py#L58-L117

However, there is a loss function applied to the bond prediction for all noise levels.

https://github.com/rssrwn/semla-flow/blob/3f43103d3af138b86dbe9f29fe8085e83f9a6283/semlaflow/models/fm.py#L850-L862

https://github.com/rssrwn/semla-flow/blob/3f43103d3af138b86dbe9f29fe8085e83f9a6283/semlaflow/models/fm.py#L757-L763

https://github.com/rssrwn/semla-flow/blob/3f43103d3af138b86dbe9f29fe8085e83f9a6283/semlaflow/models/fm.py#L591-L592

In this case, the charge prediction seems to act as a classifier applied to the last denoising step, rather than as a flow-matching model. Training the classifier on all timesteps maybe improves the robustness, even if it is only meant to be evaluated on the last timestep. Could the authors provide some insight into why the charges were handled differently than the other discrete variables?

Thanks!

**Comment by rssrwn** [comment]
Hi! Yeah, your summary is correct - the formal charges do not participate in the "flow", but are directly predicted by the model (only the final prediction is used). Note the output of the sampling procedure is model's final prediction for all data types, including charges. The reasoning behind this is that (formal) charges are much simpler than the other data types; they are nearly always zero, and they can almost always be easily inferred from the other data - atom types and bond orders.

**Comment by mattrmd** [comment]
Ok, that makes sense to me. Thanks for your response!
